### PR TITLE
FEATURE: Provide Uris in MenuItemsImplementation

### LINKED ISSUE #3907: FEATURE: Provide Uris in MenuItemsImplementation
### Description

The `MenuItemsImplementation` should already provide actual Uris additional to the provided nodes.
This way the result can directly be used with presentational components instead of requiring a recursive mapping to generate the Uris.
It might be necessary to introduce a few new variables to enable absolute Uris and maybe other settings.

### Possible Solution

Add uri to generated nested arrays in `MenuItemsImplementation`

### PR DESCRIPTION
**Pull request by Sebobo**
With this change the uris for nodes are included in menuitems. This makes it easier to use them in presentational components as no further processing is needed to turn nodes into uris.

Resolves: #3907

**Review instructions**

Debugdump the output of a `Neos.Neos:MenuItems` instance. It should include `uri` fields for all items.

**Checklist**

- [x] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [x] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [x] Reviewer - The first section explains the change briefly for change-logs
- [x] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Review by jonnitto — APPROVED** [review]
Tested it and it works as expected. Thank you for this change ❤️

**Comment by Sebobo** [comment]
Note: This might need a tiny bit more time to generate as we create all Uris. Some menus might not render items (breadcrumbs, see #1438) and skip these.

**Review by kdambekalns — COMMENTED** [review]
Seems useful to me.

Though it could lead to unneeded rendering, if people use the nodes to render URIs on their own…

Which makes me wonder, if we would need to adjust aynthing in the default rendering of Neos? And there should be something in the documentation pointing out the pre-rendered URIs, I think.

**Comment by Sebobo** [comment]
> Which makes me wonder, if we would need to adjust aynthing in the default rendering of Neos? And there should be something in the documentation pointing out the pre-rendered URIs, I think.

Thx, updated the docs and templates. Shouldn't be breaky IMO

File: Neos.Neos/Resources/Private/Templates/FusionObjects/BreadcrumbMenu.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
 					{item.label}
 				</f:then>
 				<f:else>
-					<neos:link.node node="{item.node}" />
+					<f:link.external uri="{item.uri}" title="{item.label}">{item.label}</f:link.external>
```
Comment (kdambekalns) [inline Neos.Neos/Resources/Private/Templates/FusionObjects/BreadcrumbMenu.html]: Feels a bit weird to see `external` in there… Feels like that conveys additional meaning, adding `target="_blank" or doing some other fancy trick.

Put differently: Would a "novice integrator" use `link.external` to link to URIs generated for nodes?

Comment (Sebobo) [reply Neos.Neos/Resources/Private/Templates/FusionObjects/BreadcrumbMenu.html]: Replaced it with simple a tags. Fluid confused my head...

**Review by kdambekalns — APPROVED** [review]
by reading

**Review by jonnitto — APPROVED** [review]

**Pull request merged**